### PR TITLE
generate an application token for every job

### PR DESCRIPTION
**Pull request by suiguoxin**
#5270 
- create application token in rest-server, get a k8s secret definition
- create a token object in db, (add a column in table Framework named tokenSecretDef...), let dbc create the secret with the def
- mount the token-secrets to initContainers & job containers
- revoke the token in dbc, (remove the token from DB) : realized in src/dbc

Test cases:
- submit a job with secrets
- check 'kubectl get secret | grep cred', you should see corresponding `configcred` and `tokencred`
- stop the job, the secrets should be deleted automatically

### CODE REVIEW

File: src/rest-server/src/models/v2/job/k8s.js [hunk 1]
```diff
@@ -442,6 +443,10 @@ const generateTaskRole = (
                   name: 'KUBE_APISERVER_ADDRESS',
                   value: apiserver.uri,
                 },
+                {
+                  name: 'REST_SERVER_URI',
+                  value: process.env.REST_SERVER_URI,
```
Comment (hzy46) [inline src/rest-server/src/models/v2/job/k8s.js:448]: Please don't use environmental variables here. Put it in `launcherConfig`

File: src/rest-server/src/models/v2/job/k8s.js [hunk 2]
```diff
@@ -1053,19 +1090,24 @@ const put = async (frameworkName, config, rawConfig) => {
...
     : null;
 
-  // generate job config secret
+  // generate the job config secret definition
   const configSecretDef = config.secrets
     ? getConfigSecretDef(frameworkName, config.secrets)
     : null;
 
+  // create an application token
+  const token = await tokenModel.create(userName, true);
```
Comment (hzy46) [inline src/rest-server/src/models/v2/job/k8s.js:1107]: What if:

```
1. the write merger is down 
2. the token is created here
3. the job cannot be submitted to write merger
```

This token will never be deleted or revoked.

Comment (hzy46) [reply src/rest-server/src/models/v2/job/k8s.js:1107]: Better to give each token a name or a label related to the job thus we can recycle them in the watchdog

File: src/rest-server/src/models/v2/job/k8s.js (hunk 1 above)
Comment (suiguoxin) [reply src/rest-server/src/models/v2/job/k8s.js]: > Please don't use environmental variables directly here. Put it in `launcherConfig`

fixed, thx

File: src/rest-server/src/models/v2/job/k8s.js (hunk 2 above)
Comment (suiguoxin) [reply src/rest-server/src/models/v2/job/k8s.js:1108]: > Better to give each token a name or a label related to the job thus we can recycle them in the watchdog (maybe implemented in the future)

Added a TODO item here, will resolve this later

**Review by hzy46 — APPROVED** [review]

**Pull request merged**